src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs

```diff
@@ -50,6 +51,21 @@ public class AdminCertificatesHandler : RequestHandler
                                                         $"'{RavenConfiguration.GetKey(x => x.Security.ClusterCertificatePath)}'/'{RavenConfiguration.GetKey(x => x.Security.ClusterCertificateExec)}'. " +
                                                         $"For a more detailed explanation please read about authentication and certificates in the RavenDB documentation.");
 
+                
+                if (PlatformDetails.RunningOnPosix)
+                {
+                    // For the client certificate to work properly, we need that the issuer (our server certificate) will be registered in the trusted root store.
+                    // In Linux, when using SslStream AuthenticateAsServer, the server sends the list of allowed CAs to the client. So the client's CA must be in the list. See RavenDB-8524
+                    using (var currentUserStore = new X509Store(StoreName.Root, StoreLocation.CurrentUser))
+                    {
+                        currentUserStore.Open(OpenFlags.ReadOnly);
+                        if (currentUserStore.Certificates.Contains(Server.ClusterCertificateHolder.Certificate) == false)
+                            throw new InvalidOperationException($"Cannot generate the client certificate '{certificate.Name}'. " +
+                                                                "First, you must register the server certificate in the trusted root store, on the server machine." +
+                                                                "This step is required because you are using a self-signed server certificate.");
```

Comment:
Include friendly name of the server certificate as well as its path if we have it.
And I don't think the code does what you think it does. It needs the cert itself to be registered, not its CA